### ISSUE TITLE
BUG: call of lr scheduler before optimizer

### ISSUE THREAD

**Issue by Xylambda** [Xylambda/torchfitter#10]
Apparently, a call to the lr_scheduler is being performed before de optimizer, even though the callback in charge of doing this is performing the action `on_train_batch_end`. Seems a bug related with the epoch counting in the training loop.

```python
 UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
```

**Comment by Xylambda** [comment]
Modifying the LearningRateScheduler callback like: 
```python
def on_train_batch_end(self, params_dict):
    if params_dict[ParamsDict.EPOCH_NUMBER] == 1:
        pass
    else:
        self.scheduler.step()
```


seems to solve the issue
